services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/MultipartS3AsyncClient.java

```diff
@@ -63,8 +64,10 @@ private MultipartS3AsyncClient(S3AsyncClient delegate, MultipartConfiguration mu
         MultipartConfigurationResolver resolver = new MultipartConfigurationResolver(validConfiguration);
         long minPartSizeInBytes = resolver.minimalPartSizeInBytes();
         long threshold = resolver.thresholdInBytes();
+        long apiCallBufferSize = resolver.apiCallBufferSize();
         mpuHelper = new UploadObjectHelper(delegate, resolver);
         copyObjectHelper = new CopyObjectHelper(delegate, minPartSizeInBytes, threshold);
+        downloadObjectHelper = new DownloadObjectHelper(delegate, apiCallBufferSize);
```

Comment:
Can we rename it `S3MultipartAsyncClient` or has that ship already sailed?

Reply:
The whole class? It's an `@SdkInternalApi`, so it should be fine